### ISSUE TITLE
assist_regrading() for team requires student ids

### ISSUE THREAD

**Issue by mdogucu** [mdogucu/gradetools#30]
As I was grading, I wanted to go back and regrade a team's specific question. The team is called "01" but I got an error

>Error in assist_regrading(rubric_path = "rubric.csv", grading_progress_log_path = "grading-progress-log.csv",  : 
  argument "students_to_regrade" is missing, with no default

I am not sure why I should provide student identifiers in this instance. Remembering team id is much easier.

**Comment by federicazoe** [comment]
@mdogucu, from the [vignette on team grading](https://federicazoe.github.io/gradetools/articles/c-extended-capability-teams.html#grading-with-assist_team_grading):

> When regrading with [assist_regrading()](https://federicazoe.github.io/gradetools/reference/assist_regrading.html) the only difference when team grading is teams_to_regrade must be provided and students_to_regrade must be set to NULL.

From the documentation on `assist_regrading()`:

> assist_regrading(
  rubric_path,
  grading_progress_log_path,
  final_grade_sheet_path,
  questions_to_regrade,
  students_to_regrade,
  teams_to_regrade = NULL,
  missing_assignment_grade = NA,
  github_issues = FALSE
)

And

> ### Arguments
>  ...
 > **teams_to_regrade**
vector of strings; This argument is for team grading (i.e. if one assignment is connected with multiple students). team_identifiers corresponding to teams to grade, or "all" to specify all assignments should be graded. All teams_to_regrade must be team_identifiers present in the roster

Based on that, you should run:

`assist_regrading(rubric_path = "rubric.csv", grading_progress_log_path = "grading-progress-log.csv", students_to_regrade = NULL, teams_to_regrade = "01")`

Can you let me know if that works?

**Comment by mdogucu** [comment]
Very interesting. This works. Perhaps we can make it a little more intuitive and should not expect the user to provide an argument that is not used.

**Comment by federicazoe** [comment]
I am glad that it worked. I do agree that we should make it more intuitive though!

**Comment by federicazoe** [comment]
I think we should replace the inputs named `student_identifier` and `team_identifier` with a single input named `submission_identifier`.
In my opinion this is an intuitive name that would fit well to both of those.

This would solve the current oddity of having to set `student_identifier = NULL` when grading teams.

In this case we also need to
- [ ] change `example_student_identifier` to be `example_submission_identifier` 
- [ ] update docs and vignettes

**Comment by CatalinaMedina** [comment]
I agree @federicazoe and @mdogucu  that sounds like a good change.

**Comment by mdogucu** [comment]
I agree with the change, however, usually changing arguments between one version of the package to the next, can break people's old code. So anyone who wrote gradetools code in the previous version should be able to use their code for a while maybe for two more versions. 

What I have seen other packages do is give deprecation warnings first. So please read about good practices for deprecation workflow first.
